### DISCUSSION TITLE
Enum const from TypeSpec?

### DISCUSSION THREAD

**Discussion by rhodee** [microsoft/typespec#4038] [Q&A]
I would like to generate a const enum of the JSON schema from the json schema emitter. What options do we currently have to do this within Typespec?

**Comment by timotheeguerin** [comment]
Do you want to do that [Playground](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCkDELMYOCm5hbWVzcGFjZSDGEXM7Cgptb2RlbCBQZXJzb24gewogIHRoaXNJc0FDb25zdDogImFiYyI7Cn0K&e=%40typespec%2Fjson-schema&options=%7B%7D) 
```tsp
import "@typespec/json-schema";

using TypeSpec.JsonSchema;

@jsonSchema
namespace Schemas;

model Person {
  thisIsAConst: "abc";
}

```

**Reply by rhodee** [reply]
Thank you for the reply @timotheeguerin. I realize the question was ambiguous. What I meant is generating an enum const from the output of the json schema emitter.

**Reply by timotheeguerin** [reply]
Not sure I fully get what you are saying, do you have an example?

**Reply by rhodee** [reply]
I've provided an example [here](https://typespec.io/playground?options=%7B%22options%22%3A%7B%22%40typespec%2Fjson-schema%22%3A%7B%22file-type%22%3A%22json%22%7D%7D%7D&c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCkDELMYOCm5hbWVzcGFjZSDGEXM7Cgptb2RlbCBQZXJzb24gewogIC8qKiBUaGUgcMUTJ3MgZmlyc3QgxDwuICovCiAgQG1pbkxlbmd0aCg1KQogIGRpc3BsYXlOYW1lOiBzdHJpbmc7CsdKx1lhZGRyZXNzxkDHDTogQcYJyS1MaXN0IG9mIG5pY2vFcccxQHVuaXF1ZUl0ZW1zxR3Ea3M%2FyG1bXdFCY2Fyc%2BcAwiBvd27HSMQWPzogQ2FyxDN9CgrEM1Jlc3ByZXNlbnQgYe0AqeYBGucApOUBG3N0cmVldMh3O8RSaXR5zRBvdW50cssTfcdHQ2Fy6QFeS2luZOcApcRoICBraW5kOiAiZXYiIHwgImljZeQB1cYsQnJhxi10aGXKMWLEFvEBbE3FatAsYOUAgWDKLn0K&e=%40typespec%2Fjson-schema):

The gist of what I want to do is:

1. Emit JSON Schema (as done in the link)
2. Generate an equivalent TS file with a const enum of the JSON schema (e.g. Cars.ts)

Is this clearer? Apologies for the back and forth.

**Reply by timotheeguerin** [reply]
I see, so mean you would have this sort of output in a TypeScript file?
```ts
const Person = {
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "$id": "Person.json",
    "type": "object",
    "properties": {
        "displayName": {
            "type": "string",
            "minLength": 5,
            "description": "The person's first name."
        },
        "address": {
            "$ref": "Address.json",
            "description": "Person address"
        },
        "nickNames": {
            "type": "array",
            "items": {
                "type": "string"
            },
            "uniqueItems": true,
            "description": "List of nick names"
        },
        "cars": {
            "type": "array",
            "items": {
                "$ref": "Car.json"
            },
            "description": "List of cars person owns"
        }
    },
    "required": [
        "displayName",
        "address"
    ]
} as const;
```

if so we don't have this functionality right now but this is something you could make a custom emitter for and use the json schema emitter to get the json schema.

**Reply by rhodee** [reply]
That output is exactly what I was seeking. What code exists that I can review that leverages an existing emitter's output to emit something else?
